superduperdb/components/datatype.py

```diff
@@ -563,74 +532,75 @@ class Artifact(_BaseEncodable):
     """
 
     leaf_type: t.ClassVar[str] = 'artifact'
+    artifact: t.ClassVar[bool] = True
     x: t.Any = Empty()
     blob: dc.InitVar[t.Optional[t.Union[str, bytes]]] = None
     lazy: t.ClassVar[bool] = False
 
     def __post_init__(self, db, blob=None):
         super().__post_init__(db)
         self._blob = blob
+        self._reference = None
+
+        # TODO: Need to sort out the initialization of the artifact
+        # 1. Use x(bytes) and schema for decoding. (bytes from _blobs)
+        # 2. Use x(reference) and schema for decoding.
+        # 3. use _blob(reference) for decoding. (bytes from _blobs)
+        # 4. use _blob(bytes) for decoding
```

Comment:
@blythed Please confirm.

Reply:
I think that `x` is the actual decoded thing (like `PIL.Image`) and `blob` is the `bytes`. `blob` can either be a reference or pure `bytes`, depending on the class.

Reply:
Done